app/builders/messages/facebook/message_builder.rb

```diff
@@ -53,7 +53,22 @@ def build_message
   end
 
   def conversation
```

Comment:
wrapping all the logic into a single block doesn't look readable. Please refactor it to independent methods. 
example 

```
def conversation
  @conversation ||= set_conversation_based_on_inbox_config
end

private

def set_conversation_based_on_inbox_config
  if @inbox.lock_to_single_conversation
    Conversation.find_by(conversation_params) || build_conversation
  else
    find_or_build_for_multiple_conversations
  end
end

def find_or_build_for_multiple_conversations
   # If lock to single conversation is disabled, we will create a new conversation if previous conversation is resolved
  last_conversation = Conversation.where(conversation_params).where.not(status: :resolved).order(created_at: :desc).first
  return build_conversation if last_conversation.nil?

  last_conversation
end
```